packages/blocks/src/api/registration.js

```diff
@@ -365,6 +366,17 @@ export function registerBlockType( blockNameOrMetadata, settings ) {
 		return;
 	}
 
+	// Handle `lock` as a "build-in" attribute.
+	// But allow blocks to specify their own with defaults.
+	if ( ! has( settings.attributes, [ 'lock', 'type' ] ) ) {
```

Comment:
I noticed while updating unit tests that there's an expectation for empty attributes if the block doesn't have them.

Should I update the condition and only add the `lock` attribute if attributes aren't empty?